src/soko_loader.rs

```diff
@@ -158,3 +178,20 @@ pub fn load_level(contents: &str) -> Result<World, String> {
         _ => Err("Level must start with a title".to_string()),
     }
 }
+
+pub fn parse_sokoban_worlds(sokoban_text: &str) -> Result<Vec<World>, String> {
+    let tokens = tokenize(sokoban_text);
+    if tokens.is_none() {
+        return Err("Level failed to load".to_string());
+    }
+    let tokens = tokens.unwrap();
```

Comment:
what about:

```
if let Ok(tokens) = tokens {
    let sokoban_token_groups = group_sokoban_tokens(tokens);
    let mut worlds = Vec::new();
    for level in sokoban_token_groups {
        if let Ok(world) = parse_sokoban_level(level.clone()) {
            worlds.push(world);
        }
    }
} else {
    Err("Message")
}
```

instead of the unwrap

Reply:
I pushed this version of the code.